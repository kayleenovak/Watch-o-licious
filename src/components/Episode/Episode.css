.seasons-heading {
  color: white;
  font-size: 22px;
  margin-bottom: 10px;
}

.series-seasons {
  max-height: 293px;
  border: 2px solid transparent;
  transition: all .2s; 
  overflow: auto
}

.series-seasons:hover {
  border: 2px solid #AEA7B9;
  transition: all .2s; 
}

.series-season {
  border: 2px solid transparent;
  background: rgba(166, 218, 211, .5);
  margin: 0;
  padding: 10px;
  transition: all .2s;
  color: white;
}

.series-season:hover {
  background: rgba(166, 218, 211, .9);
  border: 2px solid #AEA7B9;
  transition: all .2s;
}

.season-selected {
  background: rgba(166, 218, 211, .9);
  border: 2px solid #AEA7B9;
  transition: all .2s;
  margin: 0;
  padding: 10px;
}

.expanded-episode {
  background: rgba(174, 167, 185);
  border: 2px solid transparent;
  border-bottom: 2px solid #442C3E;
}

.collapsed-episode {
  height: 80px;
  background: rgba(174, 167, 185, .5);
  border: 2px solid transparent;
  border-bottom: 2px solid #AEA7B9;
}

.collapsed-episode:hover,
.expanded-episode:hover {
  border: 2px solid #d2d1cf;
  transition: all .5s;
}

.collapsed-episode:hover {
  background: rgba(174, 167, 185);
  transition: all .5s;
}

.episode-title {
  margin: 0;
  padding: 5px;
  color: white;
  font-family: 'Open Sans', sans-serif;
  margin-bottom: 5px;
}

.episode-info {
  color: white;
  padding: 10px;
  font-size: 14px;
}

.episode-summary {
  color: white;
}

.arrow {
  height: 20px;
  cursor: pointer;
  position: absolute;
  top: -30px;
  right: 10px;
  transition: all .4s;
}

.arrow-up {
  transform: rotate(180deg);
  height: 20px;
  cursor: pointer;
  position: absolute;
  top: -30px;
  right: 10px;
  transition: all .4s;
}

.episode-btn-wrapper {
  width: 100%;
  display: flex;
  justify-content: space-between;
  position: relative;
}

.watchlist-button,
.watched-button,
.favorite-button {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background:linear-gradient(to bottom, #442c3e 5%, #947b8e 100%);
  background-color:#a6d4d3;
  border-radius:2px;
  border:2px solid #442c3e;
  display:inline-block;
  cursor:pointer;
  color:#ffffff;
  font-family:Arial;
  font-size:14px;
  font-weight:bold;
  padding:6px 6px;
  text-decoration:none;
  text-shadow:0px 1px 1px #442C3E;
}

.favorite-button {
  margin-left: 10px;
}

.watchlist-button {
  margin-right: 10px;
}

.watchlist-button,
.watched-button,
.tracked-watchlist,
.tracked-watched {
  width: 180px;
}

.watchlist-button:hover,
.watched-button:hover,
.favorite-button:hover {
  background:linear-gradient(to bottom, #947b8e 5%, #442c3e 100%);
  background-color:#769190;
}

.watchlist-button:active,
.watched-button:active,
.favorite-button:active {
  position: relative;
  top: 1px;
}

.favorite-button {
  width: 100px;
}

.tracked-favorite,
.tracked-watched,
.tracked-watchlist {
  background: #442c3e;
  border-radius:2px;
  border:2px solid #442c3e;
  display:inline-block;
  cursor:pointer;
  color:#ffffff;
  font-family:Arial;
  font-size:14px;
  font-weight:bold;
  padding:6px 6px;
  text-decoration:none;
  text-shadow:0px 1px 1px #442C3E;
}

.tracked-favorite {
  width: 100px;
  margin-left: 10px;
}

.tracked-watchlist {
  margin-right: 10px;
}

.tracked-favorite:hover {
  border: 2px solid white;
}


