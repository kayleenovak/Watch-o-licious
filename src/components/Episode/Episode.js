import React, { Component } from 'react'

export default class Episode extends Component {
  constructor() {
    super()
    this.state = {
      expanded: false
    }
  }

  expandEpisode = () => {
    this.setState({
      expanded: !this.state.expanded
    })
  }

  handleTrackedEpisode = (e) => {
    const matchedEpisode = this.props.tracked.find(episode => {
      return this.props.url === episode.url
    })
    if(matchedEpisode === false) {
      this.props.addToTracked(this.props.episode)
      this.handleToggleTracked(e)
    } else {
      this.handleToggleTracked(e)
    }
  }

  handleToggleTracked = (e) => {
    if (e.target.value === 'favorite') {
      this.props.toggleFavorite(this.props.episode)
    } else if (e.target.value === 'watched') {
      this.props.toggleWatched(this.props.episode)
    } else if (e.target.value === 'watchlist') {
      this.props.toggleWatchList(this.props.episode)
    }
    this.props.handleRemoveTracked()
  }

  handleRemoveTracked = () => {
    const { favorite, watchlist, watched } = this.props.episode.tracked
    if (favorite === false && watchlist === false &&& watched === false) {
      this.props.removeFromTracked(this.props.episode)
    }
  }

  render() {
    const { episode, runtime, summary, title, airdate } = this.props.episode
    return (
      <section className='episode' onClick={() => this.expandEpisode()}>
        <h3>Episode {episode}: {title}</h3>
        {
          !this.state.expanded ? null : <div><p>{summary}</p><p>Runtime: {runtime} minutes</p><p>Original airdate: {airdate}</p><button value='favorite'>Favorite</button><button value='watchlist'>Watch List</button><button value='watched'>Watched</button></div>
        }
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  tracked: state.tracked
})

export const mapDispatchToProps = (dispatch) => ({
  addToTracked: (episode) => dispatch(addToTracked(episode)),
  removeFromTracked: (episode) => dispatch(removeFromTracked(episode)),
  toggleFavorite: (episode) => dispatch(toggleFavorite(episode)),
  toggleWatched: (eipsode) => dispatch(toggleWatched(episode)),
  toggleWatchList: (episode) => dispatch(toggleWatchList(episode))
})