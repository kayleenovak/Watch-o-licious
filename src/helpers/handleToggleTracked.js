import { addToTracked, removeFromTracked, toggleFavorite, toggleWatched, toggleWatchList } from '../actions/index.js'

export const handleTrackedEpisode = (e, episode, tracked) => {
  return (dispatch) => {
    const matchedEpisode = tracked.find(currentEpisode => {
      return episode.url === currentEpisode.url
    })
    if(matchedEpisode === undefined) {
      dispatch(addToTracked(episode))
      dispatch(handleToggleTracked(e, episode, tracked))
    } else {
      dispatch(handleToggleTracked(e, episode, tracked))
    }
  }
}

export const handleToggleTracked = (e, episode, tracked) => {
  const target = e.target.value
  return (dispatch) => {
    if (target === 'favorites') {
      dispatch(toggleFavorite(episode))
    } else if (target === 'watched') {
      dispatch(toggleWatched(episode))
    } else if (target === 'watchlist') {
      dispatch(toggleWatchList(episode))
    }
    dispatch(handleRemoveTracked(tracked))
  }
}

export const handleRemoveTracked = (tracked, episode) => {
  return (dispatch) => {
    const { favorites, watchlist, watched } = tracked
    if (favorite === false && watchlist === false && watched === false) {
      dispatch(removeFromTracked(episode))
    }
  }
}