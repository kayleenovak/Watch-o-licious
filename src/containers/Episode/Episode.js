import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { handleTrackedEpisode } from '../../helpers/handleToggleTracked.js'
import './Episode.css'

export class Episode extends Component {
  constructor() {
    super()
    this.state = {
      expanded: false
    }
  }

  expandEpisode = () => {
    this.setState({
      expanded: !this.state.expanded
    })
  }

  render() {
    if(this.props.episode) {
      const { episode, runtime, summary, title, airdate, tracked } = this.props.episode
      const episodeStyle = this.state.expanded ? 'expanded-episode' : 'collapsed-episode' 
      const arrow = this.state.expanded ? 'arrow-up' : 'arrow'
      const watched = tracked.watched ? 'Remove from Watched' : 'Add to Watched'
      const watchlist = tracked.watchlist ? 'Remove from Watch List' : 'Add to Watch List'
      const favoriteText = tracked.favorites ? 'Unfavorite' : 'Favorite'

      return (
        <section className={episodeStyle}>
          <h3 className='episode-title'>Episode {episode}: {title}</h3>
          <div className='episode-btn-wrapper'>
            <button className={tracked.favorites ? 'tracked-favorite' : 'favorite-button'} onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.trackedEpisodes)} value='favorite'>{favoriteText}</button>
            <button className={tracked.watched ? 'tracked-watched' : 'watched-button'} onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.trackedEpisodes)} value='watched'>{watched}</button>
            <button className={tracked.watchlist ? 'tracked-watchlist' : 'watchlist-button'} onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.trackedEpisodes)} value='watchlist'>{watchlist}</button>
          <img alt='expand-arrow' src='/down-arrow.svg' className={arrow} onClick={() => this.expandEpisode()}/>
          </div>
          {
            !this.state.expanded ? null : <div className='episode-info'><p className='episode-summary'>{summary}</p><p>Runtime: {runtime} minutes</p><p>Original airdate: {airdate}</p></div>
          }
        </section>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  trackedEpisodes: state.tracked
})

export const mapDispatchToProps = (dispatch) => ({
  handleTracked: (e, episode, tracked) => dispatch(handleTrackedEpisode(e, episode, tracked)) 
})

export default connect(mapStateToProps, mapDispatchToProps)(Episode)

Episode.propTypes = {
  episode: PropTypes.object.isRequired,
  trackedEpisodes: PropTypes.array,
  handleTracked: PropTypes.func.isRequired 
}
