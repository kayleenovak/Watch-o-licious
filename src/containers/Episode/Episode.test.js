import React from 'react'
import { shallow } from 'enzyme'
import { Episode, mapDispatchToProps, mapStateToProps } from './Episode.js'
import { handleTrackedEpisode } from '../../helpers/handleToggleTracked'

jest.mock('../../helpers/handleToggleTracked')

describe('Episode', () => {
  let wrapper
  let mockHandleTracked
  let mockEpisode
  let mockTrackedShows

  beforeEach(() => {
    mockHandleTracked = jest.fn()
    mockEpisode = {
      showId: 6407,
      url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
      title: 'Holiday Cookie Madness',
      episode: 1,
      runtime: 60,
      summary: 'This is a summary',
      airdate: '2014-11-09',
      tracked: {
        favorites: false,
        watched: false,
        watchlist: false
      }
    }
    mockTrackedShows = [{
      showId: 6407,
      url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
      title: 'Holiday Cookie Madness',
      episode: 1,
      runtime: 60,
      summary: 'This is a summary',
      airdate: '2014-11-09',
      tracked: {
        favorites: false,
        watched: false,
        watchlist: false
      }
    }]
    wrapper = shallow(<Episode 
      handleTracked={ mockHandleTracked }
      trackedEpisodes={ mockTrackedShows }
      episode={ mockEpisode }
    />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })
  
  it('should return a section, h3, button(3), div(1), img(1) if the state is false', () => {
    expect(wrapper.find('section').length).toEqual(1)
    expect(wrapper.find('button').length).toEqual(3)
    expect(wrapper.find('h3').length).toEqual(1)
    expect(wrapper.find('div').length).toEqual(1)
    expect(wrapper.find('img').length).toEqual(1)
  })

  it('should return a section, h3, div(2), img(1) and two p elements if the state is true', () => {
    wrapper.instance().expandEpisode()

    expect(wrapper.find('section').length).toEqual(1)
    expect(wrapper.find('button').length).toEqual(3)
    expect(wrapper.find('h3').length).toEqual(1)
    expect(wrapper.find('div').length).toEqual(2)
    expect(wrapper.find('p').length).toEqual(3)
    expect(wrapper.find('img').length).toEqual(1)
  })

  it('should invoke handleTracked on click of the favorite, watched, and watchlist buttons', () => {
    const mockEvent = {
      target: {
        value: 'favorite'
      }
    }

    wrapper.find('.favorite-button').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpisode, mockTrackedShows)
  })

  it('should invoke handleTracked on click of the favorite, watched, and watchlist buttons', () => {
    const mockEvent = {
      target: {
        value: 'watched'
      }
    }

    wrapper.find('.watched-button').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpisode, mockTrackedShows)
  })

  it('should invoke handleTracked on click of the favorite, watched, and watchlist buttons', () => {
    const mockEvent = {
      target: {
        value: 'watchlist'
      }
    }

    wrapper.find('.watchlist-button').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpisode, mockTrackedShows)
  })

  describe('expandEpisode', () => {
    let wrapper
    let mockHandleTracked
    let mockTrueEpisode
    let mockTrueTrackedShows

    beforeEach(() => {
      mockHandleTracked = jest.fn()
      const mockEpisode = {
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: false,
          watched: false,
          watchlist: false
        }
      }
      mockTrueEpisode = {
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: true,
          watched: true,
          watchlist: true
        }
      }
      const mockTrackedShows = [{
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: false,
          watched: false,
          watchlist: false
        }
      }]
      mockTrueTrackedShows = [{
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: true,
          watched: true,
          watchlist: true
        }
      }]
      wrapper = shallow(<Episode 
        handleTracked = { mockHandleTracked }
        tracked={ mockTrackedShows }
        episode={ mockEpisode }
      />)
    })
    it('expandEpisodshould toggle the state of expanded', () => {
      const wrapper = shallow(<Episode handleTracked = { mockHandleTracked } episode={ mockEpisode } />)

      wrapper.instance().expandEpisode()

      expect(wrapper.state().expanded).toEqual(true)

      wrapper.instance().expandEpisode()

      expect(wrapper.state().expanded).toEqual(false)
    })

    it('should invoke expandEpisode on click', () => {
      const spyExpandEpisode = jest.spyOn(wrapper.instance(), 'expandEpisode')
      wrapper.instance().forceUpdate()

      wrapper.find('.arrow').simulate('click')

      expect(spyExpandEpisode).toHaveBeenCalled()
    })

    it('should render add to watched in the watched button if tracked.watched is false', () => {

      expect(wrapper.find('.watched-button').text()).toEqual('Add to Watched')
    })

    it('should render remove from watched in the watched button if tracked.watched is true', () => {
      const trueWrapper = shallow(<Episode episode={ mockTrueEpisode } tracked={ mockTrueTrackedShows } handleTracked={ mockHandleTracked }/>)
      
      expect(trueWrapper.find('.tracked-watched').text()).toEqual('Remove from Watched')
    })

    it('should render add to watch list in the watch list button if tracked.watchlist is false', () => {

      expect(wrapper.find('.watchlist-button').text()).toEqual('Add to Watch List')
    })

    it('should render remove from watched in the watched button if tracked.watched is true', () => {
      const trueWrapper = shallow(<Episode episode={ mockTrueEpisode } tracked={ mockTrueTrackedShows } handleTracked={ mockHandleTracked }/>)
      
      expect(trueWrapper.find('.tracked-watchlist').text()).toEqual('Remove from Watch List')
    })


  })

  describe('mapStateToProps', () => {
    it('should return an object with a key of tracked', () => {
      const mockState = {
        isLoading: false, 
        hasErrored: false,
        tracked: []
      }
      const expected = {
        trackedEpisodes: []
      }

      const result = mapStateToProps(mockState)

      expect(result).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn()
      const mockEvent = {
        target: {
          value: 'favorite'
        }
      }
      const mockEpisode = {
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: false,
          watched: false,
          watchlist: false
        }
      }
      const mockTrackedShows = [{
        showId: 6407,
        url: 'http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness',
        title: 'Holiday Cookie Madness',
        episode: 1,
        runtime: 60,
        summary: 'This is a summary',
        airdate: '2014-11-09',
        tracked: {
          favorites: false,
          watched: false,
          watchlist: false
        }
      }]
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.handleTracked(mockEvent, mockEpisode, mockTrackedShows)

      expect(mockDispatch).toHaveBeenCalledWith(handleTrackedEpisode(mockEvent, mockEpisode, mockTrackedShows))
    })
  })
})