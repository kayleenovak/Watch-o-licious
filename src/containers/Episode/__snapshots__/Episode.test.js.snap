// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Episode
    episode={
      Object {
        "airdate": "2014-11-09",
        "episode": 1,
        "runtime": 60,
        "showId": "6407",
        "summary": "This is a summary",
        "title": "Holiday Cookie Madness",
        "tracked": Object {
          "favorites": false,
          "watched": false,
          "watchlist": false,
        },
        "url": "http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness",
      }
    }
    handleTracked={[MockFunction]}
    tracked={
      Array [
        Object {
          "airdate": "2014-11-09",
          "episode": 1,
          "runtime": 60,
          "showId": "6407",
          "summary": "This is a summary",
          "title": "Holiday Cookie Madness",
          "tracked": Object {
            "favorites": false,
            "watched": false,
            "watchlist": false,
          },
          "url": "http://www.tvmaze.com/episodes/377894/holiday-baking-championship-1x01-holiday-cookie-madness",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="episode-title"
        >
          Episode 
          1
          : 
          Holiday Cookie Madness
        </h3>,
        <div
          className="episode-btn-wrapper"
        >
          <button
            className="favorite-button"
            onClick={[Function]}
            value="favorite"
          >
            Favorite
          </button>
          <button
            className="watched-button"
            onClick={[Function]}
            value="watched"
          >
            Add to Watched
          </button>
          <button
            className="watchlist-button"
            onClick={[Function]}
            value="watchlist"
          >
            Add to Watch List
          </button>
          <img
            alt="expand-arrow"
            className="arrow"
            onClick={[Function]}
            src="/down-arrow.svg"
          />
        </div>,
        null,
      ],
      "className": "collapsed-episode",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Episode ",
            1,
            ": ",
            "Holiday Cookie Madness",
          ],
          "className": "episode-title",
        },
        "ref": null,
        "rendered": Array [
          "Episode ",
          1,
          ": ",
          "Holiday Cookie Madness",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorite-button"
              onClick={[Function]}
              value="favorite"
            >
              Favorite
            </button>,
            <button
              className="watched-button"
              onClick={[Function]}
              value="watched"
            >
              Add to Watched
            </button>,
            <button
              className="watchlist-button"
              onClick={[Function]}
              value="watchlist"
            >
              Add to Watch List
            </button>,
            <img
              alt="expand-arrow"
              className="arrow"
              onClick={[Function]}
              src="/down-arrow.svg"
            />,
          ],
          "className": "episode-btn-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "favorite-button",
              "onClick": [Function],
              "value": "favorite",
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add to Watched",
              "className": "watched-button",
              "onClick": [Function],
              "value": "watched",
            },
            "ref": null,
            "rendered": "Add to Watched",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add to Watch List",
              "className": "watchlist-button",
              "onClick": [Function],
              "value": "watchlist",
            },
            "ref": null,
            "rendered": "Add to Watch List",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "expand-arrow",
              "className": "arrow",
              "onClick": [Function],
              "src": "/down-arrow.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      null,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="episode-title"
          >
            Episode 
            1
            : 
            Holiday Cookie Madness
          </h3>,
          <div
            className="episode-btn-wrapper"
          >
            <button
              className="favorite-button"
              onClick={[Function]}
              value="favorite"
            >
              Favorite
            </button>
            <button
              className="watched-button"
              onClick={[Function]}
              value="watched"
            >
              Add to Watched
            </button>
            <button
              className="watchlist-button"
              onClick={[Function]}
              value="watchlist"
            >
              Add to Watch List
            </button>
            <img
              alt="expand-arrow"
              className="arrow"
              onClick={[Function]}
              src="/down-arrow.svg"
            />
          </div>,
          null,
        ],
        "className": "collapsed-episode",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Episode ",
              1,
              ": ",
              "Holiday Cookie Madness",
            ],
            "className": "episode-title",
          },
          "ref": null,
          "rendered": Array [
            "Episode ",
            1,
            ": ",
            "Holiday Cookie Madness",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorite-button"
                onClick={[Function]}
                value="favorite"
              >
                Favorite
              </button>,
              <button
                className="watched-button"
                onClick={[Function]}
                value="watched"
              >
                Add to Watched
              </button>,
              <button
                className="watchlist-button"
                onClick={[Function]}
                value="watchlist"
              >
                Add to Watch List
              </button>,
              <img
                alt="expand-arrow"
                className="arrow"
                onClick={[Function]}
                src="/down-arrow.svg"
              />,
            ],
            "className": "episode-btn-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "className": "favorite-button",
                "onClick": [Function],
                "value": "favorite",
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add to Watched",
                "className": "watched-button",
                "onClick": [Function],
                "value": "watched",
              },
              "ref": null,
              "rendered": "Add to Watched",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add to Watch List",
                "className": "watchlist-button",
                "onClick": [Function],
                "value": "watchlist",
              },
              "ref": null,
              "rendered": "Add to Watch List",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "expand-arrow",
                "className": "arrow",
                "onClick": [Function],
                "src": "/down-arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
